evm/src/executor/inspector/cheatcodes/ext.rs

```diff
@@ -281,13 +281,29 @@ fn remove_file(state: &mut Cheatcodes, path: impl AsRef<Path>) -> Result<Bytes,
 /// it will call itself to convert each of it's value and encode the whole as a
 /// Tuple
 fn value_to_token(value: &Value) -> Result<Token, Token> {
+    println!("value to token: {:?}", value);
     if value.is_boolean() {
         Ok(Token::Bool(value.as_bool().unwrap()))
     } else if value.is_string() {
         let val = value.as_str().unwrap();
         // If it can decoded as an address, it's an address
         if let Ok(addr) = H160::from_str(val) {
             Ok(Token::Address(addr))
+        } else if val.starts_with("0x") {
+            println!("val {val}");
+            // If incorrect length, pad 0 at the beginning
+            if val.len() % 2 == 0 {
+                let arr = val.strip_prefix("0x").unwrap();
+                // if length == 32 bytes, then encode as Bytes32, else Bytes
+                Ok(if arr.len() == 64 {
+                    Token::FixedBytes(Vec::from_hex(arr).unwrap())
+                } else {
+                    Token::Bytes(Vec::from_hex(arr).unwrap())
+                })
+            } else {
+                let arr = format!("0{}", val.strip_prefix("0x").unwrap());
+                Ok(Token::Bytes(Vec::from_hex(arr).unwrap()))
+            }
```

Comment:
there's `str::strip_prefix` which returns an option, so this should be an `if let Some() val.strip_prefix("0x")`